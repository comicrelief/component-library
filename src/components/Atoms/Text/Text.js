import React from 'react';
import styled, { css } from 'styled-components';
import PropTypes from 'prop-types';

/**
 * Text component
 */
export const BaseText = styled.span`
  color: ${({ color, theme }) =>
    color ? theme.color(color) : 'inherit'};
  font-size: ${({ size, theme }) => theme.fontSize(size)};
`;

/** Text renders different elements based on the `tag` prop */
const Text = ({ tag, children, ...rest }) => (
  <BaseText {...rest} as={tag}>
    {children}
  </BaseText>
);

Text.defaultProps = {
  tag: 'span',
  weight: 'normal',
  uppercase: false,
  size: 's0'
};

Text.propTypes = {
  /** Weight of Font */
  weight: PropTypes.string,
  /** Sets text transform to uppercase. */
  uppercase: PropTypes.bool,
  /** Colors */
  color: PropTypes.string,
  /** Tag type */
  tag: PropTypes.string,
  /** Sizes */
  size: PropTypes.string
};

export default Text;
