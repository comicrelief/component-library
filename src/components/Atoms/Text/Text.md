```js
<div>
  <Text tag="h3" size="s">
    Heading 3
  </Text>
  <Text tag="p" colour="black">
    My paragraph
  </Text>
  <Text tag="h3" size="m">
    Heading 3 size 3
  </Text>
  <Text tag="p" size="xl" color="red">
    My paragraph small and yellow
  </Text>
</div>
```

Sport Relief

```js
  <div>
    <Text tag="h3" size="s">
      Heading 3
    </Text>
    <Text tag="p" colour="black">
      My paragraph
    </Text>
    <Text tag="h3" size="m">
      Heading 3 size 3
    </Text>
    <Text tag="p" size="xl" color="red">
      My paragraph small and yellow
    </Text>
  </div>
```
