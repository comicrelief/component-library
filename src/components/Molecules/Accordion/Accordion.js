import React, { useState, useEffect, useRef } from 'react';
import PropTypes from 'prop-types';
import styled, { css, keyframes } from 'styled-components';

import spacing from '../../../theme/shared/spacing';
import { Chevron } from '../../Atoms/Icons/index';

const Container = styled.div`
  position: relative;
  border-radius: 1rem;
  background: ${({ theme }) => theme.color('white')};
  box-shadow: 0px 0px 20px rgba(0, 0, 0, 0.15);
  padding: ${spacing('l')} ${spacing('xxl')} ${spacing('l')} ${spacing('xl')};
`;

const ChevAnima = keyframes`
 0% { margin-top: 0rem; }
 50% { margin-top: 0.2rem; }
 100% { margin-top: 0rem; }
`;

const Button = styled.button`
  background: none;
  border: 2px solid ${({ theme }) => theme.color('red')};
  position: absolute;
  bottom: -1rem;
  right: calc(${spacing('m')} * 2);
  width: 48px;
  height: 48px;
  border-radius: 50%;
  background: ${({ theme }) => theme.color('red')};
  transition: bottom 0.1s linear;
  &:focus,
  &:hover {
    outline: none;
    border: 2px solid ${({ theme }) => theme.color('red_dark')};
    > div {
      animation-name: ${ChevAnima};
      animation-duration: 0.4s;
    }
  }
`;

const Icon = styled.div`
  display: flex;
  justify-content: center;
  align-content: center;
`;

const Copy = styled.div`
  overflow: hidden;
  visibility: none;
  transition: all 0.2s cubic-bezier(0.21, 1.7, 0.83, 0.68) 0s;
  ${({ isOpen }) => (isOpen && css`
    visibility: visible;
    margin-top: ${spacing('l')};
    transition: all 0.2s cubic-bezier(0.21, 1.7, 0.83, 0.68) 0s;
  `)}
`;

const Accordion = ({ children, title }) => {
  const [height, setHeight] = useState('auto');
  const [isOpen, setIsOpen] = useState(false);
  const stageCanvasRef = useRef(null);
  const handleOpen = () => {
    setIsOpen(!isOpen);
  };

  useEffect(() => {
    if (stageCanvasRef.current) {
      setHeight(stageCanvasRef.current.getBoundingClientRect().height);
    }
  }, [height]);

  return (
    <Container>
      {title}
      <Copy style={{ height: isOpen ? 'auto' : '0' }} ref={stageCanvasRef} isOpen={isOpen}>{children}</Copy>
      <Button onClick={handleOpen} aria-expanded={isOpen ? 'true' : 'false'}>
        <Icon>
          <Chevron direction={isOpen ? 'up' : 'down'} />
        </Icon>
      </Button>
    </Container>
  );
};

Accordion.propTypes = {
  children: PropTypes.node.isRequired,
  title: PropTypes.node.isRequired
};

export default Accordion;
