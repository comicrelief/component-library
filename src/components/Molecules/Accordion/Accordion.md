# Accordion

```js
import Text from '../../Atoms/Text/Text';

<Accordion title="I am a heading">
 <Text tag="p">Name, surname, email and billing address We need these to process your payment, create a receipt and send it to you. Establishment information We use this information to understand which institutions (e.g. schools, companies) raise money for us. Your details will be kept safe and never shared with other organisations; see our Privacy Policy for more information</Text>
</Accordion>
```
