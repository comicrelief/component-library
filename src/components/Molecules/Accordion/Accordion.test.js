import React from 'react';
import 'jest-styled-components';
import renderWithTheme from '../../../../tests/hoc/shallowWithTheme';
import Accordion from './Accordion';
import Text from '../../Atoms/Text/Text';

it('renders correctly', () => {
  const tree = renderWithTheme(
    <Accordion
      title={(
        <Text tag="h2" font="Anton" size="l">
          I am the title
        </Text>
      )}
    >
      <Text tag="p">Name, surname, email and billing address We need these to process your payment, create a receipt and send it to you. Establishment information We use this information to understand which institutions (e.g. schools, companies) raise money for us. Your details will be kept safe and never shared with other organisations; see our Privacy Policy for more information</Text>
    </Accordion>
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
