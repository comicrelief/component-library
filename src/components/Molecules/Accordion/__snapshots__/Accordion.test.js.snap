// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
.c2 {
  font-size: 1.5rem;
  line-height: 1.5rem;
  text-transform: inherit;
  font-family: 'Montserrat',Helvetica,Arial,sans-serif;
}

.c6 {
  font-size: 1rem;
  line-height: 1rem;
  text-transform: inherit;
  line-height: normal;
  font-family: 'Montserrat',Helvetica,Arial,sans-serif;
}

.c4 {
  -webkit-transform: rotate(180deg);
  -ms-transform: rotate(180deg);
  transform: rotate(180deg);
  fill: #000000;
}

.c0 {
  border-radius: 1rem;
  background: #FFFFFF;
  box-shadow: 0px 0px 20px rgba(0,0,0,0.15);
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  background: none;
  border: none;
  -webkit-transition: bottom 0.1s linear;
  transition: bottom 0.1s linear;
  cursor: pointer;
  text-align: left;
  padding: 2rem;
}

.c1:focus,
.c1:hover {
  color: inherit;
  outline: none;
}

.c1:focus > div,
.c1:hover > div {
  -webkit-animation-name: jiZjjg;
  animation-name: jiZjjg;
  -webkit-animation-duration: 0.4s;
  animation-duration: 0.4s;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-content: center;
  -ms-flex-line-pack: center;
  align-content: center;
}

.c5 {
  overflow: hidden;
  height: 0;
  visibility: none;
  -webkit-transition: all 0.2s cubic-bezier(0.21,1.7,0.83,0.68) 0s;
  transition: all 0.2s cubic-bezier(0.21,1.7,0.83,0.68) 0s;
  padding: 0 2rem;
}

@media (min-width:1024px) {
  .c4 {
    fill: #000000;
  }
}

@media (min-width:740px) {
  .c1 {
    padding: 2rem 3rem;
  }
}

@media (min-width:740px) {
  .c5 {
    padding: 0 3rem;
  }
}

<div
  className="c0"
>
  <button
    aria-expanded="false"
    className="c1"
    onClick={[Function]}
  >
    <h2
      className="c2"
      color="inherit"
      size="l"
    >
      I am the title
    </h2>
    <div
      className="c3"
    >
      <svg
        className="c4"
        direction="down"
        fill="#000000"
        height={24}
        viewBox="0 0 48 48"
        width={24}
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M40 33a1.992 1.992 0 01-1.414-.586L24 17.829 9.414 32.414a2 2 0 11-2.828-2.828l16-16a2 2 0 012.828 0l16 16A2 2 0 0140 33z"
        />
      </svg>
    </div>
  </button>
  <div
    className="c5"
  >
    <p
      className="c6"
      color="inherit"
      size="s"
    >
      Name, surname, email and billing address We need these to process your payment, create a receipt and send it to you. Establishment information We use this information to understand which institutions (e.g. schools, companies) raise money for us. Your details will be kept safe and never shared with other organisations; see our Privacy Policy for more information
    </p>
  </div>
</div>
`;
